File: algorithms/binary_search/binary_search.py
def binarySaerch(list,item):
    
    low = 0
    high = len(list) - 1
    
    while low <= high:
        midPoint = (low + high) // 2 #Integer division for midpoint
        
        guess = list[midPoint]
        
        if guess == item: 
            return midPoint # the position of the item
        
        if guess < item : # Item is in upper half of the list
            # increase the low to the next value after the midpoint value
            low = midPoint + 1
        else :
            # Item is in lower half of the list
            
            # decrease high to the value before the midpoint value
            
            high = midPoint - 1
        
    return None # Item not found



print(binarySaerch([1,3,4,5,7,9,45],5))

File: algorithms/binary_search/binary_search.java
package algorithms.binary_search;

public class binary_search {

    int binarySearch(int[] list, int item) {

        int low = 0;
        int high = list.length - 1;

        while (low <= high) {
            int midPoint = low + (high - low) / 2; // To prevent potential overflow.

            int guess = list[midPoint];

            if (guess == item) {
                return midPoint; // Return the index where the item is found.
            }

            if (guess < item) { // If the guess is less than the item,
                low = midPoint + 1; // set low to one after the midpoint.
            } else { // If the guess is greater than the item,
                high = midPoint - 1; // set high to one before the midpoint.
            }
        }

        return -1; // Return -1 if the item is not found.
    }

}
